lib/commands/project/project.js

```diff
@@ -422,53 +422,53 @@ function testGenerate(directory, options, cb) {
-          return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.testModule, TESTMODULES.join(', '))));
-        }
-
-        // check if the assertion-format is one of the three
-        if (options.assertionFormat && !_.includes(ASSERTIONTYPES, options.assertionFormat)) {
-          return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.assertionFormat, ASSERTIONTYPES.join(', '))));
-        }
-
-        // process the paths option
-        if (options.pathName){
-          var reg = new RegExp(options.pathName);
-          desiredPaths = pathList.filter(function(val) {
-            return val.match(reg);
-          });
-        }
-
-        // pass the config to the module and get the result string array
-        var config = {
-          pathName: desiredPaths,
-          testModule: testModule,
-          assertionFormat: assertionFormat
-        };
-
-        var finalResult = template.testGen(result, config);
-
-        //output the result
-        async.eachSeries(finalResult, function (item, cb) {
-          var p = item.name;
-          var filename = p.substring(1);
-          fs.outputFile(path.join(directory + '/test', filename), item.test, cb);
-        }, cb);
 
+    //read the yaml file and validate it
+    readProject(directory, options, function(err, project) {
+      if (err) { return cb(err); }
+      swaggerSpec.validateSwagger(project.api.swagger, options, function(err) {
+        spec.resolve(project.api.swagger, function(err, result) {
+          // get the array of string paths from json object
+          pathList = Object.keys(result.paths);
+
+          //check if the test frame is one of the two
+          if (options.testModule && !_.includes(TESTMODULES, options.testModule)) {
+            return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.testModule, TESTMODULES.join(', '))));
+          }
+
+          // check if the assertion-format is one of the three
+          if (options.assertionFormat && !_.includes(ASSERTIONTYPES, options.assertionFormat)) {
+            return cb(new Error(util.format('Unknown type: %j. Valid types: %s', options.assertionFormat, ASSERTIONTYPES.join(', '))));
+          }
+
+          // process the paths option
+          if (options.pathName){
+            var reg = new RegExp(options.pathName);
+            desiredPaths = pathList.filter(function(val) {
+              return val.match(reg);
+            });
+          }
+
+          // pass the config to the module and get the result string array
+          var config = {
+            pathName: desiredPaths,
+            testModule: testModule,
+            assertionFormat: assertionFormat
+          };
+          var finalResult = template.testGen(result, config);
+
+          //output the result
+          async.each(finalResult, function (item, callback) {
+            var p = item.name;
+            var filename = p.substring(1);
```

Comment:
b/c in template module, it just replace all the / with -, so the filename would look like -user-test.js, and in swagger-node, I just substring from the sencond, so it will be user-test.js, but I can move this to template module, i think moving to template module might be better?